src/general/IWorldEffect.cs

```diff
@@ -34,71 +36,177 @@ public void OnRegisterToWorld()
 
     public void OnTimePassed(double elapsed, double totalTimePassed)
     {
-        var glucoseDefinition = SimulationParameters.GetCompound(Compound.Glucose);
-
-        var totalAmount = 0.0f;
-        var totalChunkAmount = 0.0f;
+        ApplyCompoundsAddition();
+        HandlePatchCompoundDiffusion();
+    }
 
-        var initialTotalDensity = 0.0f;
-        var finalTotalDensity = 0.0f;
+    private void ApplyCompoundsAddition()
```

Comment:
This effect no longer just reduces the glucose, so calling this `GlucoseReductionEffect` is completely incorrect in my mind.

Also this is so long that I think the new effect (called maybe something like `CompoundProductionEffect`) should be put in its separate file.